lib/lsp-server/Cargo.toml

```diff
@@ -9,7 +9,8 @@ edition = "2021"
 [dependencies]
 log = "0.4.17"
 serde_json = "1.0.96"
-serde = { version = "1.0.156", features = ["derive"] }
+# See https://github.com/serde-rs/serde/issues/2538#issuecomment-1684517372 for why we pin serde
+serde = { version = "1.0.156, < 1.0.171", features = ["derive"] }
```

Comment:
I think there is a mistake and it should be `< 1.0.172` like on https://github.com/rust-lang/rust-analyzer/pull/15481/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R101

Reply:
@matklad :point_up:

Reply:
Yeah, looks like 1.0.171 is the latest version without it.